go-controller/pkg/csrapprover/approver.go

```diff
@@ -78,51 +132,45 @@ var Predicate = predicate.Funcs{
 
 // NewController creates a new OVNKubeCSRController
 func NewController(client crclient.Client,
-	commonNamePrefixes string,
-	organization []string,
-	groups, userPrefixes sets.Set[string],
+	csrAcceptanceConditions []CSRAcceptanceCondition,
 	usages sets.Set[certificatesv1.KeyUsage],
 	maxDuration time.Duration,
 	recorder record.EventRecorder) *OVNKubeCSRController {
-	c := &OVNKubeCSRController{
-		name:               ControllerName,
-		client:             client,
-		commonNamePrefixes: commonNamePrefixes,
-		organization:       organization,
-		groups:             groups,
-		usages:             usages,
-		userPrefixes:       userPrefixes,
-		maxDuration:        maxDuration,
-		recorder:           recorder,
-	}
-	return c
-}
 
-func (c *OVNKubeCSRController) filterCSR(csr *certificatesv1.CertificateSigningRequest) bool {
-	nsn := types.NamespacedName{Namespace: csr.Namespace, Name: csr.Name}
-	csrPEM, _ := pem.Decode(csr.Spec.Request)
-	if csrPEM == nil {
-		klog.Errorf("Failed to PEM-parse the CSR block in .spec.request: no CSRs were found in %s", nsn)
-		return false
+	commonNamePrefixes := []string{}
+
+	// create commonNamePrefixes from csrAcceptanceConditions
+	for _, v := range csrAcceptanceConditions {
+		commonNamePrefixes = append(commonNamePrefixes, v.CommonNamePrefix)
 	}
 
-	x509CSR, err := x509.ParseCertificateRequest(csrPEM.Bytes)
-	if err != nil {
-		klog.Errorf("Failed to parse the CSR .spec.request of %q: %v", nsn, err)
-		return false
+	return &OVNKubeCSRController{
+		name:                    ControllerName,
+		client:                  client,
+		csrAcceptanceConditions: csrAcceptanceConditions,
+		commonNamePrefixes:      commonNamePrefixes,
+		usages:                  usages,
+		maxDuration:             maxDuration,
+		recorder:                recorder,
 	}
+}
 
-	return strings.HasPrefix(x509CSR.Subject.CommonName, c.commonNamePrefixes) &&
-		csr.Spec.SignerName == certificatesv1.KubeAPIServerClientSignerName
+func (c *OVNKubeCSRController) filterCSR(csr *certificatesv1.CertificateSigningRequest, x509CSR *x509.CertificateRequest) bool {
+	for _, v := range c.commonNamePrefixes {
```

Comment:
Is there a point of having the `commonNamePrefixes` field in `OVNKubeCSRController`?
Or can we just iterate over csrAcceptanceConditions and check `CommonNamePrefix`?

Reply:
`commonNamePrefixes` has all commonNamePrefix in csrAcceptanceCondition at `func NewController(...) *OVNKubeCSRController`. https://github.com/ovn-org/ovn-kubernetes/pull/3909/files/217d1e5384867c776d3bedbb0dc673ef83a0728e#diff-9c5c1788aa52e78e5d64bb6dd82c4a16dc432bd9c8fb69f79f04b286746e003aR142-R144
 So if `filterCSR()` is invoked before `NewController()` it is invalid, but I suppose we never have such situation (because filterCSR() is a method of OVNKubeCSRController).